sdk/python/feast/infra/online_stores/contrib/cassandra_online_store/cassandra_online_store.py

```diff
@@ -0,0 +1,525 @@
+                exe_profile = ExecutionProfile(load_balancing_policy=lb_policy)
+                execution_profiles = {EXEC_PROFILE_DEFAULT: exe_profile}
+            else:
+                execution_profiles = None
+
+            # additional optional keyword args to Cluster
+            cluster_kwargs = {
+                k: v
+                for k, v in {
+                    "protocol_version": protocol_version,
+                    "execution_profiles": execution_profiles,
+                }.items()
+                if v is not None
+            }
+
+            # creation of Cluster (Cassandra vs. Astra)
+            if hosts:
+                self._cluster = Cluster(
+                    hosts, port=port, auth_provider=auth_provider, **cluster_kwargs
+                )
+            else:
+                # we use 'secure_bundle_path'
+                self._cluster = Cluster(
+                    cloud={"secure_connect_bundle": secure_bundle_path},
+                    auth_provider=auth_provider,
+                    **cluster_kwargs,
+                )
+
+            # creation of Session
+            self._keyspace = keyspace
+            self._session = self._cluster.connect(self._keyspace)
+
+        return self._session
+
+    def __del__(self):
+        """
+        One may be tempted to reclaim resources and do, here:
+            if self._session:
+                self._session.shutdown()
+        But *beware*, DON'T DO THIS.
+        Indeed this could destroy the session object before some internal
+        tasks runs in other threads (this is handled internally in the
+        Cassandra driver).
+        You'd get a RuntimeError "cannot schedule new futures after shutdown".
+        """
+        pass
+
+    @log_exceptions_and_usage(online_store="cassandra")
+    def online_write_batch(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        data: List[
+            Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        """
+        Write a batch of features of several entities to the database.
+
+        Args:
+            config: The RepoConfig for the current FeatureStore.
+            table: Feast FeatureView.
+            data: a list of quadruplets containing Feature data. Each
+                  quadruplet contains an Entity Key, a dict containing feature
+                  values, an event timestamp for the row, and
+                  the created timestamp for the row if it exists.
+            progress: Optional function to be called once every mini-batch of
+                      rows is written to the online store. Can be used to
+                      display progress.
+        """
+        project = config.project
+        #
+        for entity_key, values, timestamp, created_ts in data:
+            entity_key_bin = serialize_entity_key(entity_key).hex()
```

Comment:
Strongly recommend using `entity_key_serialization_version=2` in `serialize_entity_key`.

Reply:
Yep, I'm glad you noticed that!